openshift-knative-operator/pkg/serving/extension_test.go

```diff
@@ -134,10 +135,24 @@ func TestReconcile(t *testing.T) {
 		expected: ks(func(ks *v1alpha1.KnativeServing) {
 			ks.Status.MarkDependenciesInstalled()
 		}),
+	}, {
+		name: "wrong namespace",
+		in: ks(func(ks *v1alpha1.KnativeServing) {
+			ks.Namespace = "foo"
+		}),
+		expected: ks(func(ks *v1alpha1.KnativeServing) {
+			ks.Namespace = "foo"
+			ks.Status.MarkInstallFailed(`Knative Serving must be installed into the namespace "knative-serving"`)
+		}),
 	}}
 
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
+			// Default the namespace to the correct one if not set for brevity.
+			if c.in.Namespace == "" {
```

Comment:
Could we also test the case where ns = ""

Reply:
That's not possible in practice, is it? Since the API is namespace scoped there always has to be a namespace set.

Reply:
My bad I was not clear, I mean the `requiredNs` to be "".